### PR TITLE
Add dimensions support to stats

### PR DESCRIPTION
**Pull request by brycekahle**
Also change to use cumulative counters for SignalFX.

### CODE REVIEW

File: metrics/transport/signalfx.go [hunk 1]
```diff
@@ -52,5 +52,14 @@ func (t *SFXTransport) Publish(m *metrics.RawMetric) error {
 		p.Dimensions[k] = asStr
 	}
 
+	switch m.Type {
+	case metrics.GaugeType:
+		p.MetricType = datapoint.Gauge
+	case metrics.TimerType:
+		fallthrough
```
Comment (rybit) [inline metrics/transport/signalfx.go]: Timers should be reported as timers right?

Comment (brycekahle) [inline metrics/transport/signalfx.go]: SignalFX doesn't have timers 😄

Comment (rybit) [inline metrics/transport/signalfx.go]: (╯°□°）╯︵ ┻━┻

I swear they said they did...but you're right.

**Comment by rybit** [comment]
lgtm

**Comment by brycekahle** [comment]
Any concerns about running a SHA256 every time a metric is recorded?

**Comment by rybit** [comment]
So I did some benchmarking on this before I added it. I am not worried. 

```
BenchmarkSignature-8         300000        4052 ns/op      1216 B/op        27 allocs/op
```

meaning that we could do around 250k signatures/second.

**Pull request merged**